src/rules/at-rule-blacklist/index.js

```diff
@@ -0,0 +1,35 @@
+import { isString } from "lodash"
+import { vendor } from "postcss"
+
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "at-rule-blacklist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (name) => `Unexpected function "${name}"`,
+})
+
+export default function (blacklist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: blacklist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+
+    root.walkAtRules(atRule => {
```

Comment:
Our new convention is to destructure properties if we use them more than once e.g. `const { name } = atRule`.

`function-blacklist` is a older rule and so isn't up to date with this convention.